client/nginx.go

```diff
@@ -400,6 +401,34 @@ func NewNginxClient(httpClient *http.Client, apiEndpoint string) (*NginxClient,
 	return &NginxClient{
 		apiEndpoint: apiEndpoint,
 		httpClient:  httpClient,
+		version:     APIVersion,
+	}, nil
+}
+
+// NewNginxClientV4 creates an NginxClient with version 4 of NGINX Plus API.
+func NewNginxClientV4(httpClient *http.Client, apiEndpoint string) (*NginxClient, error) {
+	version := 4
+	versions, err := getAPIVersions(httpClient, apiEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("error accessing the API: %v", err)
+	}
+
+	found := false
+	for _, v := range *versions {
+		if v == version {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return nil, fmt.Errorf("API version %v of the client is not supported by API versions of NGINX Plus: %v", version, *versions)
+	}
+
+	return &NginxClient{
+		apiEndpoint: apiEndpoint,
+		httpClient:  httpClient,
+		version:     version,
 	}, nil
 }
```

Comment:
There's a lot of duplicated code in this new function, which I think could be extracted into a helper function.
Would the code below work? All existing users of this code would continue to call `NewNginxClient` and get a client for the latest version, and anyone wishing to use V4 or other could call `NewNginxClientWithVersion`?

```suggestion
// NewNginxClient creates an NginxClient with the latest version.
func NewNginxClient(httpClient *http.Client, apiEndpoint string) (*NginxClient, error) {
	return NewNginxClientWithVersion(httpClient, apiEndpoint, APIVersion)
}

// NewNginxClientWithVersion creates an NginxClient with the given version of NGINX Plus API.
func NewNginxClientWithVersion(httpClient *http.Client, apiEndpoint string, version int) (*NginxClient, error) {
	versions, err := getAPIVersions(httpClient, apiEndpoint)
	if err != nil {
		return nil, fmt.Errorf("error accessing the API: %v", err)
	}

	found := false
	for _, v := range *versions {
		if v == version {
			found = true
			break
		}
	}

	if !found {
		return nil, fmt.Errorf("API version %v of the client is not supported by API versions of NGINX Plus: %v", version, *versions)
	}

	return &NginxClient{
		apiEndpoint: apiEndpoint,
		httpClient:  httpClient,
		version:     version,
	}, nil
}
```

Reply:
@mikestephen This suggestion looks good.

I might suggest un-exporting `NewNginxClientWithVersion`, as this would allow creating api clients for apiVersions 1,2,3 which may have additional unknown unsupported fields/methods.

Then we could still have:
```go
// NewNginxClientV4 creates an NginxClient with version 4 of NGINX Plus API.
func NewNginxClientV4(httpClient *http.Client, apiEndpoint string) (*NginxClient, error) {
	return newNginxClientWithVersion(httpClient, apiEndpoint, 4)
}
```

Supporting just apiVersion 4 and 5 will add support for [R18](https://docs.nginx.com/nginx/releases/#r18)+, as far as I know. Which seems reasonable.

Reply:
Same thoughts... PR will be updated shortly.

Reply:
Updated, also I extracted the error and made it exported, for convenience.

Reply:
The final version has the exported `NewNginxClientWithVersion()` and check for supported versions.